contracts/solana/stork/programs/stork/src/lib.rs

```diff
@@ -0,0 +1,244 @@
+use anchor_lang::prelude::*;
+use anchor_lang::solana_program::system_instruction;
+
+mod verify;
+use verify::{verify_stork_evm_signature, EvmPubkey};
+
+declare_id!("F8C2ffLfHGQTQxnix4LaroXuHh2Cdv8wJw7R9UDEfmX1");
+
+pub const STORK_CONFIG_SEED: &[u8] = b"stork_config";
+pub const STORK_FEED_SEED: &[u8] = b"stork_feed";
+pub const STORK_TREASURY_SEED: &[u8] = b"stork_treasury";
+
+#[program]
+pub mod stork {
+    use super::*;
+
+    pub fn initialize(
+        ctx: Context<Initialize>,
+        stork_sol_public_key: Pubkey,
+        stork_evm_public_key: EvmPubkey,
+        single_update_fee_in_lamports: u64,
+    ) -> Result<()> {
+        let config = &mut ctx.accounts.config;
+        config.stork_sol_public_key = stork_sol_public_key;
+        config.stork_evm_public_key = stork_evm_public_key;
+        config.single_update_fee_in_lamports = single_update_fee_in_lamports;
+        config.owner = ctx.accounts.owner.key();
+        Ok(())
+    }
+
+    pub fn update_temporal_numeric_value_evm(
+        ctx: Context<UpdateTemporalNumericValue>,
+        update_data: TemporalNumericValueEvmInput,
+    ) -> Result<()> {
+        let config = &ctx.accounts.config;
+        let treasury = &ctx.accounts.treasury;
+        let feed = &mut ctx.accounts.feed;
+        let payer = &ctx.accounts.payer;
+
+        if feed.id == [0; 32] {
+            feed.id = update_data.id;
+        }
+
+        if !verify_stork_evm_signature(
+            &config.stork_evm_public_key,
+            update_data.id,
+            update_data.temporal_numeric_value.timestamp_ns,
+            update_data.temporal_numeric_value.quantized_value,
+            update_data.publisher_merkle_root,
+            update_data.value_compute_alg_hash,
+            update_data.r,
+            update_data.s,
+            update_data.v,
+        ) {
+            return Err(StorkError::InvalidSignature.into());
```

Comment:
Should probably switch to use the anchor err! macro. This should have the same result but be more idiomatic.